extensions/stats/plugin.h

```diff
@@ -399,6 +403,27 @@ class PluginContext : public Context {
     rootCtx->report(request_info_);
   };
 
+  void onDownstreamConnectionClose(PeerType) override {
+    downstream_closed_ = true;
+    if (upstream_closed_ && !tcp_reported_) {
+      tcp_reported_ = true;
+      auto rootCtx = rootContext();
+      ::Wasm::Common::populateTCPRequestInfo(rootCtx->outbound(),
+                                             &request_info_);
+      rootCtx->report(request_info_);
+    }
+  }
+  void onUpstreamConnectionClose(PeerType) override {
+    upstream_closed_ = true;
+    if (downstream_closed_ && !tcp_reported_) {
+      tcp_reported_ = true;
```

Comment:
Dedupe the code between the two callbacks.

Reply:
Done..